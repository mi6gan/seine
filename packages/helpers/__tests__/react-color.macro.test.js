const pluginTester = require('babel-plugin-tester').default;
const plugin = require('babel-plugin-macros');

pluginTester({
  plugin,
  babelOptions: {
    plugins: [require('@babel/plugin-transform-react-jsx').default],
    filename: __filename,
  },
  snapshot: true,
  tests: [
    `
      // @flow
      import { SketchPicker } from '../react-color.macro';

      const DefaultPicker = SketchPicker;

      export default () => <DefaultPicker />;
      export const Picker = () => {
        const Picker = SketchPicker;
        return <Picker />;
      }
      export { SketchPicker };
    `,
  ],
});
