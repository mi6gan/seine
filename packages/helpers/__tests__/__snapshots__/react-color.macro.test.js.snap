// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros 1. macros: 1. macros 1`] = `

// @flow
import { SketchPicker } from '../react-color.macro';

const DefaultPicker = SketchPicker;

export default () => <DefaultPicker />;
export const Picker = () => {
  const Picker = SketchPicker;
  return <Picker />;
}
export { SketchPicker };

      ↓ ↓ ↓ ↓ ↓ ↓

// @flow
const DefaultPicker = SketchPicker;
export default () => /*#__PURE__*/ React.createElement(DefaultPicker, null);
export const Picker = () => {
  const Picker = SketchPicker;
  return /*#__PURE__*/ React.createElement(Picker, null);
};
export { SketchPicker };


`;
